Document/0x06g-Testing-Network-Communication.md

```diff
@@ -61,46 +61,112 @@ The following is a summarised list of App Transport Security Requirements<sup>[1
+| -----| ------------|
+| `NSIncludesSubdomains` | Indicates whether ATS exceptions should apply to subdomains of the named domain |
+| `NSExceptionAllowsInsecureHTTPLoads` | Allows HTTP connections to the named domain, but does not affect TLS requirements |
+| `NSExceptionMinimumTLSVersion` | Allows connections to servers with TLS versions less than 1.2 |
+| `NSExceptionRequiresForwardSecrecy` | Disable perfect forward secrecy (PFS) |
+
+
+Starting from January 1 2017, Apple App Store review and requires justification if one of the following ATS exceptions are defined. However this decline is extended later by Apple stating “To give you additional time to prepare, this deadline has been extended and we will provide another update when a new deadline is confirmed”<sup>[5]</sup>
+
+* `NSAllowsArbitraryLoads`
+* `NSExceptionAllowsInsecureHTTPLoads`
+* `NSExceptionMinimumTLSVersion`
 
--- TODO: Describe ATS exceptions --
 
 #### Static Analysis
 
-— TODO —
+If the source code is available, open then `Info.plist` file in the application bundle directory using a text editor and look for any exceptions that the application developer has configured. This file should be examined taking the applications context into consideration. 
 
-#### Dynamic Analysis
+The following listing is an example of an exception configured to disable ATS restrictions globally. 
 
-— TODO —
+```
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoads</key>
+		<true/>
+	</dict>
+```
+
+If the source code is not available, then the `Info.plist` file should be either can be obtained from a jailbroken device or by extracting the application IPA file.
+
+Since IPA files are ZIP archives, they can be extracted using any zip utility.
+
+```
+$ unzip app-name.ipa
+```
+
+`Info.plist` file can be found in the `Payload/BundleName.app/` directory of the extract. It’s a binary encoded file and has to be converted to a human readable format for the analysis. 
+
+`plutil`<sup>[6]</sup> is a tool that’s designed for this purpose. It comes natively with Mac OS 10.2 and above versions.
+
+The following command shows how to convert the Info.plist file into XML format.
+```
+$ plutil -convert xml1 Info.plist
+```
+
+Once the file is converted to a human readable format, the exceptions can analysed. The application may have ATS exceptions defined to allow it’s normal functionality. For an example, the Firefox iOS application has ATS disabled globally. This exception is acceptable because otherwise the application would not be able to connect to any HTTP web sites or website that do not have the ATS requirements.
 
 #### Remediation
 
-— TODO —
+* If the application connects to a defined number of domains that the application author controls, then configure the servers to support the ATS requirements
+* If the application connects to a set of domains that the application owner controls and some third-party domains, then opt in for ATS support for the own domains
```

Comment:
Please change to: "If connections to 3rd party domains are made (that are not under control of the app owner) it should be evaluated what ATS settings are not supported by the 3rd party domain and deactivated. "